src/connectors/injective/injective.ts

```diff
@@ -0,0 +1,198 @@
+import {
+  MsgBatchUpdateOrders,
+  IndexerGrpcSpotApi,
+  Orderbook,
+  SpotOrderHistory,
+  GrpcOrderType,
+  spotPriceToChainPriceToFixed,
+  spotQuantityToChainQuantityToFixed,
+} from '@injectivelabs/sdk-ts';
+import {
+  ClobMarketsRequest,
+  ClobOrderbookRequest,
+  ClobTickerRequest,
+  ClobGetOrderRequest,
+  ClobPostOrderRequest,
+  ClobDeleteOrderRequest,
+  CLOBMarkets,
+  ClobGetOrderResponse,
+} from '../../clob/clob.requests';
+import { NetworkSelectionRequest } from '../../services/common-interfaces';
+import { InjectiveCLOBConfig } from './injective.clob.config';
+import { Injective } from '../../chains/injective/injective';
+import LRUCache from 'lru-cache';
+import { getInjectiveConfig } from '../../chains/injective/injective.config';
+
+export class InjectiveCLOB {
+  private static _instances: LRUCache<string, InjectiveCLOB>;
+  private _chain;
+  public conf;
+  public spotApi: IndexerGrpcSpotApi;
+  private _ready: boolean = false;
+  public parsedMarkets: CLOBMarkets = {};
+
+  private constructor(_chain: string, network: string) {
+    this._chain = Injective.getInstance(network);
+    this.conf = InjectiveCLOBConfig.config;
+    this.spotApi = new IndexerGrpcSpotApi(this._chain.endpoints.indexer);
+  }
+
+  public static getInstance(chain: string, network: string): InjectiveCLOB {
+    if (InjectiveCLOB._instances === undefined) {
+      const config = getInjectiveConfig(network);
+      InjectiveCLOB._instances = new LRUCache<string, InjectiveCLOB>({
```

Comment:
Why is `LRUCache` needed to instantiate an Injective instance, whereas other connectors like Uniswap dont?

Reply:
The cache in the case helps to control the number of instances that'll be held in memory.
It's not really a big deal considering the most likely use case by most users.
It can be removed if requested.

Reply:
The LRUCache is first added here, but really, all classes that manage instances using the class attribute dictionary approach are exposed to memory leaks and can incorporate it. As Victor pointed out, this is likely not an issue in practice, but this is a proposal to mitigate the edge cases. The con here, of course, is the additional dependency.

Reply:
Thanks guys! I'll add `LRUCache` to Uniswap connector then